test/services/address/index.unit.js

```diff
@@ -2429,6 +2433,51 @@ describe('Address Service', function() {
     });
   });
 
+
+  describe('#_updateAddressIndex', function() {
+    it('should add using 2 keys', function() {
+      var as = new AddressService({
+          mempoolMemoryIndex: true,
+          node: mocknode
+      });
+
+      _.values(as.mempoolAddressIndex).should.deep.equal([]);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index2', true);
+      as._updateAddressIndex('index2', true);
+      as.mempoolAddressIndex.should.deep.equal({
+        "index1": 4,
+        "index2": 2
+      });
+    });
+
+    it('should add/remove using 2 keys', function() {
+      var as = new AddressService({
+          mempoolMemoryIndex: true,
+          node: mocknode
+      });
+      _.values(as.mempoolAddressIndex).should.deep.equal([]);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', true);
+      as._updateAddressIndex('index1', false);
+
+      as._updateAddressIndex('index2', true);
+      as._updateAddressIndex('index2', true);
+      as._updateAddressIndex('index2', false);
+      as._updateAddressIndex('index2', false);
+      as._updateAddressIndex('index2', false);
```

Comment:
After removing index2 twice the key should be removed otherwise the index will grow indefinitely, until the process is restarted.